packages/providers/src/fuel-graphql-subscriber.ts

```diff
@@ -0,0 +1,75 @@
+import { FuelError } from '@fuel-ts/errors';
+import type { DocumentNode } from 'graphql';
+import { print } from 'graphql';
+
+type FuelGraphQLSubscriberOptions = {
+  url: string;
+  query: DocumentNode;
+  variables?: Record<string, unknown>;
+  fetchFn: typeof fetch;
+  abortController?: AbortController;
+};
+
+class FuelSubscriptionStream implements TransformStream {
+  readable: ReadableStream<FuelError | Record<string, unknown>>;
+  writable: WritableStream<Uint8Array>;
+  private readableStreamController!: ReadableStreamController<FuelError | Record<string, unknown>>;
+  private static textDecoder = new TextDecoder();
+
+  constructor() {
+    this.readable = new ReadableStream({
+      start: (controller) => {
+        this.readableStreamController = controller;
+      },
+    });
+
+    this.writable = new WritableStream<Uint8Array>({
+      write: (bytes) => {
+        const text = FuelSubscriptionStream.textDecoder.decode(bytes);
+        // the fuel node sends keep-alive messages that should be ignored
+        if (text.startsWith('data:')) {
+          const { data, errors } = JSON.parse(text.split('data:')[1]);
+          if (Array.isArray(errors)) {
+            this.readableStreamController.enqueue(
+              new FuelError(
+                FuelError.CODES.INVALID_REQUEST,
+                errors.map((err) => err.message).join('\n\n')
+              )
+            );
+          } else this.readableStreamController.enqueue(data);
```

Comment:
Yep, me too.

I'd vote for a conventional `else { .. }` block to keep consistency with the rest of the codebase.

Reply:
The codebase itself is not consistent on this. Setting `curly: ['error', 'all']` in `.eslintrc.js` gives 36 errors as of the current master. This is because our eslint config extends `airbnb-base` where they define [`curly: ['error', 'multiline']`](https://github.com/airbnb/javascript/blob/11ab37144b7f846f04f64a29b5beb6e00d74e84b/packages/eslint-config-airbnb-base/rules/best-practices.js#L31).